test/cluster/object_store/test_basic.py

```diff
@@ -240,3 +241,19 @@ async def test_memtable_flush_retries(manager: ManagerClient, tmpdir, s3_server)
     res = cql.execute(f"SELECT * FROM {ks}.{cf};")
     have_res = { x.name: x.value for x in res }
     assert have_res == dict(rows), f'Unexpected table content: {have_res}'
+
+@pytest.mark.asyncio
+async def test_get_object_store_endpoints(manager: ManagerClient, s3_server):
+    objconf = MinioServer.create_conf(s3_server.address, s3_server.port, s3_server.region)
+    badconf = MinioServer.create_conf('a', 123, 'bad_region')
+    cfg = {'object_storage_endpoints': objconf + badconf}
+
+    print('Scylla returns the object storage endpoints')
+    server = await manager.server_add(config=cfg)
+    endpoints = await manager.api.get_config(server.ip_addr, 'object_storage_endpoints')
+
+    print('Also check the returned string is valid JSON')
+    del objconf[0]['name']
+    del badconf[0]['name']
+    assert json.loads(endpoints[s3_server.address]) == objconf[0]
+    assert json.loads(endpoints['a']) == badconf[0]
```

Comment:
That's a good place to check that `select from system.config` also returns valid endpoints json

Reply:
And that updating endpoints in config makes real effect

Reply:
Start scylla with some endpoint, update scylla.yaml and kill -HUP to scylla, check that `storage_manager::_s3_endpoints` sees the updated values.

The latter, however, is _not_ to be checked by reading the endpoints from system.config or API, because it will read the data from db::config, not from storage_manager. The proper way to check is to CREATE KEYSPACE with the endpoint in question, when keyspace is created the storage_manager::is_known_endpoint()  is called.

Reply:
> That's a good place to check that select from system.config also returns valid endpoints json

It's already there (below) so I withdraw this part of the comment :)